src/t_string.c

```diff
@@ -787,17 +787,17 @@ void lcsCommand(client *c) {
         goto cleanup;
     }
 
-    /* Detect string truncation or later overflows. */
-    if (sdslen(a) >= UINT32_MAX-1 || sdslen(b) >= UINT32_MAX-1) {
-        addReplyError(c, "String too long for LCS");
-        goto cleanup;
-    }
-
     /* Compute the LCS using the vanilla dynamic programming technique of
      * building a table of LCS(x,y) substrings. */
     uint32_t alen = sdslen(a);
     uint32_t blen = sdslen(b);
 
+    /* Detect string truncation or later overflows. */
+    if (alen >= UINT32_MAX-1 || blen >= UINT32_MAX-1) {
```

Comment:
Please notice that sdslen() returns `size_t`, if we vetify here, they will be truncated.
btw: sdslen() is very fast, it saves the length of string inside.

Reply:
Oh, thanks.